NickelHook.h

```diff
@@ -63,6 +65,14 @@ struct nh_dlsym {
     bool       optional; // default: false - prevents a failure to resolve the sym from being treated as a fatal error, and sets it to NULL instead
 };
 
+// nh_delete_* deletes the provided path if it exists. It performs some checks
+// to detect and avoid deleting critical system files. The functions return true
+// if the path was removed or does not exist, and false on error. nh_delete_dir will only
+// delete empty directories. Use nh_delete_file and/or nh_delete_dir to delete any
+// files or directories within before attempting to delete a directory.
```

Comment:
line wrapping

Reply:
You want `'nh_delete_dir will only`... to start on a new line? Is that your concern?

Reply:
I keep the total line length (including spaces and slashes at the beginning) for comments to 80 chars.

Reply:
What a great extension.

Done